Packs/AWS-SecurityHub/Integrations/AWSSecurityHubEventCollector/AWSSecurityHubEventCollector.py

```diff
@@ -0,0 +1,205 @@
+import datetime
+
+import demistomock as demisto
+from AWSApiModule import *
+
+import urllib3
+import boto3
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+VENDOR = 'AWS'
+PRODUCT = 'Security Hub'
+
+
+def get_events(client: boto3.client, start_time: datetime | None = None,
+               end_time: datetime | None = None, limit: int = 0) -> list[dict]:
+    """
+    Fetch events from AWS Security Hub.
+
+    Args:
+        client (boto3.client): Boto3 client to use.
+        start_time (datetime | None, optional): Start time to fetch events from. Required if end_time is set.
+        end_time (datetime | None, optional): Time to fetch events until. Defaults to current time.
+        limit (int): Maximum number of events to fetch. Can only be values between 0-100. Defaults to 0 (no limit).
+
+    Returns:
+        tuple[list, CommandResults]: A tuple containing the events and the CommandResults object.
+
+    """
+    kwargs = {}
+    _filters = {}
+
+    if end_time and not start_time:
+        raise ValueError('start_time must be set if end_time is used.')
+
+    if start_time:
+        _filters['UpdatedAt'] = [{
+            'Start':
+                start_time.strftime('%Y-%m-%dT%H:%M:%SZ'),
+            'End':
+                end_time.strftime('%Y-%m-%dT%H:%M:%SZ') or datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
+        }]
+
+    if limit:
+        kwargs['MaxResults'] = limit
+
+    if _filters:
+        # We send kwargs because passing Filters=None to get_findings() tries to use a None value for filters,
+        # which raises an error.
+        kwargs = {'Filters': _filters}
+
+    response = client.get_findings(**kwargs)
+    events = response['Findings']
+
+    while 'NextToken' in response:
+        kwargs['NextToken'] = response['NextToken']
+
+        response = client.get_findings(**kwargs)
+        events.extend(response['Findings'])
+
+    return events
+
+
+def fetch_events(client: boto3.client, last_run: dict[str, int],
+                 first_fetch_time: datetime | None, ) -> (dict[str, int], list):
+    """
+    Fetch events from AWS Security Hub.
+
+    Args:
+        client (boto3.client): Boto3 client to use.
+        last_run (dict): A dict with a key containing the latest event created time we got from last fetch.
+        first_fetch_time (int | None, optional): If last_run is None (first time we are fetching),
+            it contains the timestamp in milliseconds on when to start fetching events.
+
+    Returns:
+        dict: Next run dictionary containing the timestamp that will be used in ``last_run`` on the next fetch.
+        list: List of events that will be generated in XSIAM.
+    """
+    if demisto.getLastRun():
+        start_time = datetime.fromtimestamp(last_run.get('lastRun', None))
+
+    else:
+        start_time = first_fetch_time
+
+    events = get_events(client, start_time)
+    last_finding_update_time = events[0].get('UpdatedAt') if events else None
```

Comment:
we always assume there are events?

Reply:
It can be an empty list, but I do check for that on `events[0].get('UpdatedAt') if events else None` if that's what you're referring to.